### ISSUE TITLE
Serializing date & time objects to ISO8601

### ISSUE THREAD

**Issue by indrekj** [ohler55/oj#499]
In v2 I was able to do:
```
irb(main):003:0> Oj::VERSION
=> "2.18.5"

irb(main):006:0> puts Oj.dump({x: Time.now, y: DateTime.now, z: Date.today}, mode: :compat, time_format: :xmlschema)
{"x":"2018-09-03T16:34:43.442962000+03:00","y":"2018-09-03T16:34:43+03:00","z":"2018-09-03"}
```

This iso8601 serialization is very good because it's super fast with oj.

The new oj version however doesn't support using time_format with `:compat` mode. So instead I tried to use `:custom` mode, but that doesn't look good either:

```
irb(main):002:0> Oj::VERSION
=> "3.6.7"

irb(main):004:0> puts Oj.dump({x: Time.now, y: DateTime.now, z: Date.today}, mode: :custom, time_format: :xmlschema)
{"x":"2018-09-03T16:36:56.380981000+03:00","y":{"s":"2018-09-03T16:36:56+03:00"},"z":{"s":"2018-09-03"}}
```

The `Time` object serialized nicely, but the `Date` and `DateTime` objects are now wrapped in `"s"` field. Can I unwrap it somehow?

I also tried using `Oj.register_odd` but was unsuccessful. I didn't understand how exactly it works.

Is it possible to achieve the initial behavior with the new version and its options?

**Comment by ohler55** [comment]
There was some reason I went with a `s` wrapper. I think it might have been to add additional DateTime fields but I suspect that did not work out so I reverted to strings. I suspect most people are like you and would prefer the `:time_format` to be honored for `Date` and `DataTime` as well. I am a bit nervous about just changing it without a way to get the old behavior so let me think a day or so and decide if an option should be added or is an override like monkey patching the classes would be best. Feel free to weigh in on that.

**Comment by ohler55** [comment]
Ahh, if `:create_additions` is true then the class name is also added to the JSON object which allows for recreation of the correct type. I think a reasonable change would be to just emit the string or what ever the `:time_format` specifies if `:create_addition` is false.

**Comment by ohler55** [comment]
Please try the `default` branch.

**Comment by ohler55** [comment]
I need to make some changes to get the timezone removed or fixed.

**Comment by indrekj** [comment]
My tests against Time and DateTime are passing 👍 . My test against Date however is failing:

```
       expect(dump(date: Date.new(2016, 1, 4)))
         .to eq('{"date":"2016-01-04"}')

       expected: "{\"date\":\"2016-01-04\"}"
            got: "{\"date\":\"2016-01-04T00:00:00+02:00\"}"
```

**Comment by ohler55** [comment]
Yeah, t4hats what I need to fix.

**Comment by ohler55** [comment]
All tests are now passing. How about for you?

**Comment by indrekj** [comment]
Nice, all tests passing here as well.

**Comment by ohler55** [comment]
Great. Released 3.6.8.

**Comment by indrekj** [comment]
Thank you!
